homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,172 @@
+    PRESET_AWAY,
+    PRESET_HOME,
+    PRESET_NONE,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+PRESET_FREEZE = "freeze"
+PRESET_NIGHT = "night"
+
+STATE_DEROGATION_ACTIVE = "active"
+STATE_DEROGATION_INACTIVE = "inactive"
+
+
+OVERKIZ_TO_HVAC_MODES: dict[str, HVACMode] = {
+    STATE_DEROGATION_ACTIVE: HVACMode.HEAT,
+    STATE_DEROGATION_INACTIVE: HVACMode.AUTO,
+}
+HVAC_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODES.items()}
+
+OVERKIZ_TO_PRESET_MODES: dict[OverkizCommandParam, str] = {
+    OverkizCommandParam.AT_HOME_MODE: PRESET_HOME,
+    OverkizCommandParam.AWAY_MODE: PRESET_AWAY,
+    OverkizCommandParam.FREEZE_MODE: PRESET_FREEZE,
+    OverkizCommandParam.MANUAL_MODE: PRESET_NONE,
+    OverkizCommandParam.SLEEPING_MODE: PRESET_NIGHT,
+    OverkizCommandParam.SUDDEN_DROP_MODE: PRESET_NONE,
+}
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+TARGET_TEMP_TO_OVERKIZ = {
+    PRESET_HOME: OverkizState.SOMFY_THERMOSTAT_AT_HOME_TARGET_TEMPERATURE,
+    PRESET_AWAY: OverkizState.SOMFY_THERMOSTAT_AWAY_MODE_TARGET_TEMPERATURE,
+    PRESET_FREEZE: OverkizState.SOMFY_THERMOSTAT_FREEZE_MODE_TARGET_TEMPERATURE,
+    PRESET_NIGHT: OverkizState.SOMFY_THERMOSTAT_SLEEPING_MODE_TARGET_TEMPERATURE,
+}
+
+# controllableName is somfythermostat:SomfyThermostatTemperatureSensor
+TEMPERATURE_SENSOR_DEVICE_INDEX = 2
+
+
+class SomfyThermostat(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Smart Thermostat."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_hvac_modes = [*HVAC_MODES_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    # Both min and max temp values have been retrieved from the Somfy Application.
+    _attr_min_temp = 15.0
+    _attr_max_temp = 26.0
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(
+            TEMPERATURE_SENSOR_DEVICE_INDEX
+        )
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODES[
+            cast(
+                str, self.executor.select_state(OverkizState.CORE_DEROGATION_ACTIVATION)
+            )
+        ]
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        if not self.current_temperature or not self.target_temperature:
+            return HVACAction.IDLE
+        if self.current_temperature < self.target_temperature:
```

Comment:
We don't allow calculating state for device or service integrating integrations. The api should return the state we set in the entity.

Reply:
Indeed. I will fix this asap.

Reply:
Done with https://github.com/home-assistant/core/pull/73776